app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,97 @@
 class SendgridApi
-  extend SingleForwardable
+  TIMEOUT = 2 # seconds
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  RESCUABLE_ERRORS = [SendgridClient::Error,
+                      JSON::ParserError,
+                      Excon::Errors::Error,
+                      Timeout::Error
+                     ].freeze
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    action = 'spamreports.get.json'
+    query = { email: email }
+
+    response = SendgridPool.instance.with { |conn|
+      conn.get_request(action, query)
+    }
+
+    return false if error_response?(response)
+
+    response.any?
+  rescue *RESCUABLE_ERRORS => e
+    Rails.logger.error("#{e.class.name}: #{e.message}")
+    false
   end
 
   def bounced?(email)
-    api { bounces.retrieve(email: email).any? }
+    action = 'bounces.get.json'
+    query = { email: email }
+
+    response = SendgridPool.instance.with { |conn|
+      conn.get_request(action, query)
+    }
+
+    return false if error_response?(response)
+
+    response.any?
+  rescue *RESCUABLE_ERRORS => e
+    Rails.logger.error("#{e.class.name}: #{e.message}")
+    false
   end
 
   def remove_from_bounce_list(email)
-    api { bounces.delete(email: email) }
+    action = 'bounces.delete.json'
+    data = { email: email }
+
+    response = SendgridPool.instance.with { |conn|
+      conn.post_request(action, data)
+    }
+
+    return false if error_response?(response)
+
+    email_removed?(response)
+  rescue *RESCUABLE_ERRORS => e
+    Rails.logger.error("#{e.class.name}: #{e.message}")
+    false
```

Comment:
If feels like there's a lot of duplication in this code which we could abstract: namely fetching connection from the pool and rescuing errors.